src/rust/engine/src/scheduler.rs

```diff
@@ -145,12 +145,13 @@ impl Session {
     *run_id = Uuid::new_v4();
   }
 
-  pub fn write_stdout(&self, msg: &str) {
+  pub async fn write_stdout(&self, msg: &str) -> Result<(), String> {
     if let Some(display) = &self.0.display {
-      let display = display.lock();
-      display.write_stdout(msg);
+      let mut display = display.lock();
+      display.write_stdout(msg).await
```

Comment:
Hm... so, it just occurred to me that another option would have been to "enqueue" stdout here and have the Display deal with it the next time that `render` was called (from the `execute` loop). Not a blocker, but maybe something we should consider doing in the future? I guess that the downside would be that it might be up to "refresh_interval" milliseconds before it actually renders.